src/components/NewItemForm.jsx

```diff
@@ -55,14 +103,111 @@ const NewItemForm = () => {
   const [endpoint, setEndpoint] = useState("");
   const [password, setPassword] = useState("");
   const [username, setUserName] = useState("");
+  const [itemNameError, setItemNameError] = useState(clearError);
+  const [endpointError, setEndpointError] = useState(clearError);
+  const [passwordError, setPasswordError] = useState(clearError);
+  const [usernameError, setUserNameError] = useState(clearError);
+
   const [collectionID, setCollectionID] = useState("");
+  const [collectionArray, setCollectionArray] = useState([]);
+  const [defaultCollectionsID, setDefaultCollectionID] = useState("");
+
+  const [symmetricKey, setSymmetricKey] = useState("");
 
   const handleCollectionChange = (event) => {
     setCollectionID(event.target.value);
   };
 
-  const onSaveVaultItem = () => {
-    console.log("status");
+  const getUserItems = async () => {
```

Comment:
This function is for retrieving collections right?

Reply:
Yes! I used getUserItems for collections and the keys. I combined it into one as I figured we'd either get both or throw an error